.github/workflows/sonarqube.yml

```
@@ -0,0 +1,42 @@
+  pull_request:
+    branches:
+      - main  # or your default branch
+
+jobs:
+  build:
+    name: Run Tests and Analyze with SonarCloud
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```

actionlint (1.7.4)

the runner of "actions/checkout@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)